src/sentry/tasks/integrations/github/open_pr_comment.py

```diff
@@ -74,3 +79,26 @@ def get_pr_filenames(
     # new files will not have sentry issues associated with them
     pr_filenames: List[str] = [file["filename"] for file in pr_files if file["status"] != "added"]
     return pr_filenames
+
+
+def get_projects_and_filenames_from_source_file(
+    org_id: int,
+    pr_filename: str,
+) -> Tuple[Set[Project], Set[str]]:
+    # fetch the code mappings in which the source_root is a substring at the start of pr_filename
+    code_mappings = (
+        RepositoryProjectPathConfig.objects.filter(organization_id=org_id)
+        .annotate(substring_match=StrIndex(Value(pr_filename), "source_root"))
+        .filter(substring_match=1)
```

Comment:
`StrIndex` returns the index in which there is a match, but uses 1-indexing. it returns 0 if there is no match

Reply:
Q: is this case-sensitive?

Reply:
i'm assuming it works like regular string indexing, but i can check

Reply:
Apparently it depends on the DB backend used. In our case it's PostgreSQL, where it's rendered as `STRPOS()`, which is indeed case-sensitive.